lib/bolt/pal.rb

```diff
@@ -360,16 +362,43 @@ def get_task(task_name)
     def list_plans(filter_content: false)
       in_bolt_compiler do |compiler|
         errors = []
-        plans = compiler.list_plans(nil, errors).map { |plan| [plan.name] }.sort
+        updated = false
+        plans = compiler.list_plans(nil, errors)
+        plan_cache = Bolt::Util.read_optional_json_file(@project&.plan_cache_file || '', 'Plan cache file')
+        plan_list = plans.map(&:name).sort.each_with_object([]) do |plan_name, pl|
+          info = plan_cache[plan_name] || get_plan_info(plan_name)
+
+          # If the plan is a 'local' plan (in the project itself, or the
+          # modules/ directory) then verify it hasn't been updated since we
+          # cached it. If it has been updated, refresh the cache and use the
+          # new data.
+          if info['file'] &&
```

Comment:
@puppetlabs/skeletor I don't think this check is useful for y'all - it seems too early to override this in [PE Pal](https://github.com/puppetlabs/orchestrator/blob/main/ruby/bolt-plan-executor-lib/bolt_plan_executor/pe_pal.rb) but I'm guessing that's the route you'll want to take.